drivers/mpsl/flash_sync/flash_sync_mpsl.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <errno.h>
+
+#include <mpsl.h>
+#include <mpsl_timeslot.h>
+#include <hal/nrf_timer.h>
+
+#include "multithreading_lock.h"
+#include "soc_flash_nrf.h"
+
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_sync_mpsl);
+
+/* The request length specified by the upper layers is only time required to do
+ * the flash operations itself. Therefore we need to add some additional slack
+ * to each timeslot request.
+ */
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+#define TIMESLOT_LENGTH_SLACK_US 1000
+#else
+#define TIMESLOT_LENGTH_SLACK_US 100
+#endif
+
+struct mpsl_context {
+	/* This semaphore is taken with a timeout when the flash operation starts. */
+	struct k_sem timeout_sem;
+	mpsl_timeslot_session_id_t session_id; /* Timeslot session ID. */
+	/* The flash operation may be split into multiple requests.
+	 * This represents the length of such a request. */
+	uint32_t request_length_us;
+	/* Argument passed to nrf_flash_sync_exe(). */
+	struct flash_op_desc *op_desc;
+	mpsl_timeslot_request_t timeslot_request;
+	/* Return parameter for the timeslot session. */
+	mpsl_timeslot_signal_return_param_t return_param;
+	int status; /* Return value for nrf_flash_sync_exe(). */
+	/* Indicate timeout condition to the timeslot callback. */
+	atomic_t timeout_occured;
+};
+
+static struct mpsl_context _context;
+
+/**
+ * Get time in milliseconds since the beginning of the timeslot.
+ *
+ * This should only be called inside the timeslot.
+ */
+static uint32_t get_timeslot_time_us(void)
+{
+	nrf_timer_task_trigger(NRF_TIMER0, NRF_TIMER_TASK_CAPTURE0);
+	return nrf_timer_cc_get(NRF_TIMER0, NRF_TIMER_CC_CHANNEL0);
+}
+
+static void reschedule_next_timeslot(void)
+{
+	_context.timeslot_request.params.earliest.priority =
+		MPSL_TIMESLOT_PRIORITY_HIGH;
+
+	int32_t ret = mpsl_timeslot_request(_context.session_id,
+					    &_context.timeslot_request);
+
+	__ASSERT_EVAL((void)ret, (void)ret, ret == 0,
+		      "mpsl_timeslot_request failed: %d", ret);
+}
+
+static mpsl_timeslot_signal_return_param_t *
+timeslot_callback(mpsl_timeslot_session_id_t session_id, uint32_t signal)
```

Comment:
```
static mpsl_timeslot_signal_return_param_t *timeslot_callback(
	mpsl_timeslot_session_id_t session_id, uint32_t signal)
```
This is usual way we deal with this.
(no action required).

Reply:
I have used `clang-format` with the settings from zephyr. It formatted the code this way.